src/ops/flat.cc

```diff
@@ -111,16 +98,18 @@ Flat::Flat(FFModel &model, const ParallelTensor _input, char const *name)
          1 /*outputs*/,
          _input) {
   assert(_input->num_dims == FlatInput::NUMDIM);
+  assert (_input->check_valid());
 
   Flat::construct_output_mappings(*this->parallel_dims_mapping);
 
-  ParallelDim output_dims[MAX_TENSOR_DIM];
-  int output_ndims;
+  ParallelTensorShape output_shape;
   this->get_params().solve_dims(
```

Comment:
Is it possible to add an overload here to take a `ParallelTensorShape`, or simply to refactor the function to that signature?

Reply:
Yes, that's much cleaner. I'll change the signature for `solve_dims()` because that is specific to each `[Op]Param` but for `model.create_parallel_tensor_legion_ordering()` I'll overload it.

Reply:
Actually, I take that back. I think overloading the model function is a bit trivial.